scheduler-server/scheduling-dsl-compiler/src/libs/scheduler-edsl-fluent-api.ts

```diff
@@ -46,6 +48,15 @@ export class Goal {
       interval: opts.interval,
     });
   }
+  public static CoexistenceGoal(opts: { activityTemplate: ActivityTemplate, forEach: string }): ActivityCoexistenceGoal {
+    return Goal.new({
+      kind: 'ActivityCoexistenceGoal',
```

Comment:
I think we should have an enum of these node kinds at this point.

```
enum NodeKind {
  ActivityCoexistenceGoal: 'ActivityCoexistenceGoal'
  ActivityRecurrenceGoal: 'ActivityRecurrenceGoal',
  ...etc
}
```

and then `kind: NodeKind.ActivityCoexistenceGoal`

Reply:
We certainly can do this for aesthetics (and I suppose to give us a nice list of all kinds) - but it won't give us any additional compile time guarantees, since `Goal.new` takes a `GoalSpecifier`, and all of the variants of GoalSpecifier specify that their `kind` field needs to be a particular string.

Reply:
Yeah, its more of an organizational thing -> easier to ensure in the future and gives some nice comparison behavior if we even need to manipulate the AST

Reply:
Turns out that importing from the AST file is broken at runtime 😅  I'm getting an `Unable to resolve dependency "./scheduler-ast.js" error. My guess is that because we were only using the types from the AST, and never values, the import statement was skipped at runtime. The NodeKind enum introduces a runtime dependency on the AST file, and the import fails.

I'll postpone adding a `NodeKind` to the AST until [[AERIE-1767]: Perform Scheduling DSL Type checking on backend](https://jira.jpl.nasa.gov/browse/AERIE-1767) is ready, since your work will allow imports to actually work at runtime :+1:

Reply:
Now that its merged, is this something we want to add in  a commit to this PR?

Reply:
I added `NodeKind` in the latest commit :+1:

Reply:
Hmm, this seems to be causing some import errors, looking into it now